lib/rhc/deployment_helpers.rb

```diff
@@ -36,76 +38,104 @@ def deploy_artifact(rest_app, artifact, hot_deploy, force_clean_build)
+        if windows?
+          ssh_cmd = "type #{filename} | #{ssh_executable} -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
+        else
+          ssh_cmd = "cat #{filename} | #{ssh_executable} -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
+        end
 
         begin
-          ssh_send_file_ruby(ssh_url.host, ssh_url.user, remote_cmd, filename)
+          if use_custom_ssh
+            run_with_system_ssh(ssh_cmd)
+          else
+            ssh_send_file_ruby(ssh_url.host, ssh_url.user, remote_cmd, filename)
+          end
           success "Success"
         rescue
-          ssh_cmd = "ssh -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
           warn "Error deploying local file. You can try to deploy manually with:\n#{ssh_cmd}"
           raise
         end
       end
 
-      def deploy_file_from_url(rest_app, file_url, hot_deploy, force_clean_build)
+      def deploy_file_from_url(rest_app, file_url, hot_deploy, force_clean_build, ssh_executable, use_custom_ssh)
         say "Deployment of file '#{file_url}' in progress for application #{rest_app.name} ..."
 
         ssh_url = URI(rest_app.ssh_url)
         file_url = URI(file_url)
 
         remote_cmd = "oo-binary-deploy#{hot_deploy ? ' --hot-deploy' : ''}#{force_clean_build ? ' --force-clean-build' : ''}"
+        ssh_cmd = "#{ssh_executable} -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
 
         begin
-          ssh_send_url_ruby(ssh_url.host, ssh_url.user, remote_cmd, file_url)
+          if use_custom_ssh
+            run_with_system_ssh(ssh_cmd)
+          else
+            ssh_send_url_ruby(ssh_url.host, ssh_url.user, remote_cmd, file_url)
+          end
           success "Success"
         rescue
-          ssh_cmd = "ssh -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
           warn "Error deploying file from url. You can try to deploy manually with:\n#{ssh_cmd}"
           raise
         end
       end
 
-      def activate_deployment(rest_app, deployment_id)
+      def activate_deployment(rest_app, deployment_id, ssh_executable, use_custom_ssh)
         say "Activating deployment '#{deployment_id}' on application #{rest_app.name} ..."
 
         ssh_url = URI(rest_app.ssh_url)
         remote_cmd = "gear activate --all #{deployment_id}"
+        ssh_cmd = "#{ssh_executable} -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
 
         begin
-          ssh_ruby(ssh_url.host, ssh_url.user, remote_cmd)
+          if use_custom_ssh
+            run_with_system_ssh(ssh_cmd)
+          else
+            ssh_ruby(ssh_url.host, ssh_url.user, remote_cmd)
+          end
           success "Success"
         rescue
-          ssh_cmd = "ssh -t #{ssh_url.user}@#{ssh_url.host} '#{remote_cmd}'"
           warn "Error activating deployment. You can try to activate manually with:\n#{ssh_cmd}"
           raise
         end
       end
 
+      def run_with_system_ssh(ssh_cmd)
```

Comment:
Maybe this could be moved to `ssh_helpers.rb` and used in `RHC::Commands::PortForward` as well.